hazelcast-jca/src/main/java/com/hazelcast/jca/ResourceAdapterImpl.java

```diff
@@ -91,9 +98,15 @@ public void endpointDeactivation(MessageEndpointFactory endpointFactory, Activat
      * @see javax.resource.spi.ResourceAdapter#start(javax.resource.spi.BootstrapContext)
      */
     public void start(BootstrapContext ctx) throws ResourceAdapterInternalException {
-        // Gets/creates the hazelcast instance
-        ConfigBuilder config = buildConfiguration();
-        hazelcastInstance = Hazelcast.newHazelcastInstance(config.build());
+        if (client != null && client) {
```

Comment:
I am not very familiar with the JCA spec so I am not sure if a null can be passed if the config property is not present in the ra.xml file

```
    <resourceadapter>
        <resourceadapter-class>com.hazelcast.jca.ResourceAdapterImpl</resourceadapter-class>
        <config-property>
            <description>Create a Hazelcast Client Instance? Defaults to false (Server)</description>
            <config-property-name>client</config-property-name>
            <config-property-type>java.lang.Boolean</config-property-type>
            <config-property-value>false</config-property-value>
        </config-property>
```

If not, then I agree that client can't be set to null.